Riot/Modules/Call/Dialpad/DialpadViewController.swift

```diff
@@ -0,0 +1,321 @@
+            
+            reformatPhoneNumber()
+            
+            phoneNumberTextField.selectedTextRange = phoneNumberTextField.textRange(from: cursorPos,
+                                                                                    to: cursorPos)
+        } else {
+            phoneNumber += digit
+            reformatPhoneNumber()
+        }
+    }
+    
+    @IBAction private func backspaceButtonAction(_ sender: DialpadActionButton) {
+        if phoneNumber.isEmpty {
+            return
+        }
+        
+        if let selectedRange = phoneNumberTextField.selectedTextRange {
+            let cursorStartPos = phoneNumberTextField.offset(from: phoneNumberTextField.beginningOfDocument, to: selectedRange.start)
+            let cursorEndPos = phoneNumberTextField.offset(from: phoneNumberTextField.beginningOfDocument, to: selectedRange.end)
+            
+            let rangePos: UITextPosition!
+            
+            if selectedRange.isEmpty {
+                //  just caret, remove one char from the cursor position
+                if cursorStartPos == 0 {
+                    //  already at the beginning of the text, no more text to remove here
+                    return
+                }
+                phoneNumber.replaceSubrange((phoneNumber.index(phoneNumber.startIndex, offsetBy: cursorStartPos-1))..<(phoneNumber.index(phoneNumber.startIndex, offsetBy: cursorEndPos)), with: "")
+                
+                rangePos = phoneNumberTextField.position(from: phoneNumberTextField.beginningOfDocument,
+                                                         offset: cursorStartPos-1)
+            } else {
+                //  really some text selected, remove selected range of text
+                
+                phoneNumber.replaceSubrange((phoneNumber.index(phoneNumber.startIndex, offsetBy: cursorStartPos))..<(phoneNumber.index(phoneNumber.startIndex, offsetBy: cursorEndPos)), with: "")
+                
+                rangePos = phoneNumberTextField.position(from: phoneNumberTextField.beginningOfDocument,
+                                                         offset: cursorStartPos)
+            }
+            
+            reformatPhoneNumber()
+            
+            guard let cursorPos = rangePos else { return }
+            phoneNumberTextField.selectedTextRange = phoneNumberTextField.textRange(from: cursorPos,
+                                                                                    to: cursorPos)
+        } else {
+            phoneNumber.removeLast()
+            reformatPhoneNumber()
+        }
+    }
+    
+    @IBAction private func callButtonAction(_ sender: DialpadActionButton) {
+        delegate?.dialpadViewControllerDidTapCall(self, withPhoneNumber: rawPhoneNumber)
+    }
+    
+}
+
+//  MARK: - CustomSizedPresentable
+
+extension DialpadViewController: CustomSizedPresentable {
```

Comment:
Maybe we could have used `preferredcontentsize ` https://developer.apple.com/documentation/uikit/uiviewcontroller/1621476-preferredcontentsize if we presented the dialpad as popover on iPad

Reply:
First, i've tried to use default presentation but had to use a custom presentation controller due to the orientation issues. So it doesn't make sense to use it now.